com.unity.netcode.gameobjects/Editor/CodeGen/NetworkBehaviourILPP.cs

```diff
@@ -737,7 +741,12 @@ private MethodReference GetFastBufferReaderReadMethod(string name, TypeReference
                             var meetsConstraints = true;
                             foreach (var constraint in method.GenericParameters[0].Constraints)
                             {
+#if UNITY_CECIL_CONSTRAINTS_ARE_TYPE_REFERENCES
```

Comment:
Curious why use the indirect directive like this instead of just doing `UNITY_2022_2_OR_NEWER`?

Reply:
Because the breaking change in Unity happened in a patch version. Unity 2022.2.0a11 needs one implementation, Unity 2022.2.0a12 needs another. I'm a little miffed about that. `UNITY_2022_2_OR_NEWER` was what I wanted to use, but they threw versioning protocol to the wind and broke a patch version so that doesn't work...

Reply:
Oh, interesting. Why are we bothering having support for different alpha versions? Shouldn't we just expect people to use the latest alpha? Or at the very least this should be cleaned up after alpha versions are irrelevant.

Maybe you guys are already thinking of this, just wanted to call it out. Thanks for explaining!

Reply:
Honestly I have no idea what's publicly available and what isn't. This just feels like the safest way to do it to me.

Reply:
alpha versions are not public, betas are.

Reply:
so, we can get away with `UNITY_2022_2_OR_NEWER` or the next whatever 2022 LTS, we don't need to cover >0.a12 here.